src/rime_api.cc

```diff
@@ -1083,6 +1083,34 @@ const char* RimeGetVersion() {
   return RIME_VERSION;
 }
 
+Bool RimeDeleteCandidate(RimeSessionId session_id, size_t index) {
+  an<Session> session(Service::instance().GetSession(session_id));
```

Comment:
avoid repeated code.

extract the common code to helper functions:
```c++
static bool do_with_candidate(RimeSessionId session_id, size_t index, bool (Context::* verb)(size_t index)) {
  // ...
  return (ctx->*verb)(index);
}

static bool do_with_candidate_on_current_page(RimeSessionId session_id, size_t index, bool (Context::* verb)(size_t index)) {
  // ...
  return (ctx->*verb)(page_start + index);
}
```
Then call the helper functions with `&Context::Select` or `&Context::Delete` in each version.

Reply:
Sorry but it turns out I have trouble using this syntax. It first throw error message with
/home/runner/work/trime/trime/app/src/main/jni/librime/src/rime_api.cc:1061:25: error: called object type 'bool 

(rime::Context::*)(size_t)' is not a function or function pointer

    return Bool(ctx->*verb(index));
                       ~~~~^
/home/runner/work/trime/trime/app/src/main/jni/librime/src/rime_api.cc:1083:53: error: use of undeclared identifier 'Select'; did you mean 'rime::Context::Select'?

    return Bool(do_with_candidate(session_id, index, &Select));
                                                      ^~~~~~
                                                      rime::Context::Select
  /home/runner/work/trime/trime/app/src/main/jni/librime/src/rime/context.h:47:8: note: 'rime::Context::Select' declared here

    bool Select(size_t index);
         ^


and then after I change `&Select` to `&rime::Context::Select` it throw error message:

home/runner/work/trime/trime/app/src/main/jni/librime/src/rime_api.cc:1061:25: error: called object type 'bool (rime::Context::*)(size_t)' is not a function or function pointer

    return Bool(ctx->*verb(index));
                      ~~~~^
  1 error generated.

Reply:
pointer to member function, if I recall correctly, need to be qualified by class name, try `&Context::Select` as in the original comment.